### ISSUE TITLE
Multi-Level Supervision ("Use All The Labels")

### ISSUE THREAD

**Issue by lowmason** [lowmason/naics-embedder#18]
**Description:**
Currently, we only contrast leaf nodes. We should treat an input text as a positive match for its entire ancestry path (Sector $\rightarrow$ Subsector $\rightarrow$ Industry Group $\rightarrow$ NAICS Industry $\rightarrow$ Industry).

**Tasks:**
- [x] Label Expansion: Modify the dataloader to return a list of ancestors [Label_2_digit, Label_2_digit, Label_4_digit, Label_5_digit, Label_6_digit] for each sample. The term Label_6_digit becomes the anchor and the positives are the other labels.
- [x] Hierarchical Triplet Construction:
 - (Anchor=Label_6_digit, Positive=Label_5_digit, Negatives=(sampled negatives))
 - (Anchor=Label_6_digit, Positive=Label_4_digit, Negatives=(sampled negatives))
 - (Anchor=Label_6_digit, Positive=Label_3_digit, Negatives=(sampled negatives))
 - (Anchor=Label_6_digit, Positive=Label_2_digit, Negatives=(sampled negatives))
- [x] Gradient Accumulation: Sum the gradients from all four sets of triplets before the backward pass.
